Created Vision package

Coronal-Halo opened the pull request:
Created a vision package. Vision package contains an image stitching node.

Coronal-Halo committed:
Created Vision package

Created a vision package. Vision package contains an image stitching node.

Coronal-Halo committed:
changed project name in CMakeLists.txt

garethellis0 commented on src/vision/package.xml:
```diff
@@ -0,0 +1,56 @@
+  <build_depend>std_msgs</build_depend>
+  <run_depend>roscpp</run_depend>
+  <run_depend>rospy</run_depend>
+  <run_depend>std_msgs</run_depend>
```
We also depend on `sensor_msgs` for this, so you should add a `build_depend` and `run_depend` for them (similar to what is already here for `std_msgs`

Coronal-Halo replied:
Fixed

Coronal-Halo replied:
And could you explain a little more about why we need to do so.

garethellis0 replied:
Sure, no problem! These tags dictate things that must be installed for certain things to happen. You can see a detailed description for these on `3.3` of [this page](http://wiki.ros.org/catkin/package.xml#Build.2C_Run.2C_and_Test_Dependencies) (note that we currently use the old "format 1" `package.xml` style). Thanks for asking! I love when people ask about things they don't know (one of the main goals of this team is learning after all!)

Coronal-Halo replied:
Thank you very much!

garethellis0 commented on src/vision/scripts/imgPublisher.py:
```diff
@@ -0,0 +1,81 @@
+	image_cv = CvBridge().cv2_to_imgmsg(result, "bgr8")
+
+	try: 
+		ImgStitcher.publish(image_cv)
```
You should test that this is actually publishing the merged image properly. To do this, you need to use `rviz`:
- Open a terminal and run `rviz`
- In `rviz`, click `Add`, then select `By Topic` and then choose the topic you're broadcasting this image on. You should be able to see a video stream.

Coronal-Halo replied:
Verified it's working

garethellis0 commented on src/vision/scripts/imgPublisher.py:
```diff
@@ -0,0 +1,81 @@
+        except CvBridgeError as e:
+		print(e)
+
+	print("Published stitched image")
```
We should remove this before we merge this. It's useful for debugging, but it's going to clutter up the output (especially if there are multiple nodes running in the same terminal later on)

Coronal-Halo replied:
Fixed

garethellis0 commented on src/vision/scripts/listener.py:
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# Software License Agreement (BSD License)
```
Ok, I think this is just an example subscriber that we're not using, so you should be good to delete it.

Coronal-Halo replied:
Removed

garethellis0 commented on src/vision/scripts/imgPublisher.py:
```diff
@@ -0,0 +1,81 @@
+total = 0
+
+ImgStitcher = rospy.Publisher('ImgStitcher', Image, queue_size=10) 
+rospy.init_node('ImgPublisher', anonymous=True)
```
We don't want anonymous=True here, as this will just spawn this node with a different name if there is already an instance of it running. If we create another instance of this node, we want the first one to automatically be killed, because we don't want two programs trying to access USB devices at the same time. Please go through the rest of this PR and remove any other instances of it as well.

Coronal-Halo replied:
Fixed

garethellis0 commented on src/vision/scripts/pyimagesearch/panorama.py:
```diff
@@ -0,0 +1,103 @@
+			# if the match is None, then there aren't enough matched
+			# keypoints to create a panorama
+			if M is None:
+				print("Cannot stitched")
```
Again, useful for debugging, but we should remove this before we merge to avoid cluttering the output

Coronal-Halo replied:
Fixed

garethellis0 commented on src/vision/scripts/talker.py:
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+# Software License Agreement (BSD License)
```
I think this is just an example file, so we should be able to delete it.

Coronal-Halo replied:
Removed

garethellis0 commented on src/vision/scripts/sampleSubscriber.py:
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
+
+import rospy
```
I think this is just an example file, so we should be able to delete it.

Coronal-Halo replied:
Removed

garethellis0 commented on src/vision/scripts/talker.py:
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+# Software License Agreement (BSD License)
+#
+# Copyright (c) 2008, Willow Garage, Inc.
```
This seems like an example file that we can delete.

Coronal-Halo replied:
Removed

garethellis0 commented on src/vision/scripts/sampleSubscriber.py:
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python
+
+import rospy
```
This seems like an example file that you can delete.

Coronal-Halo replied:
Removed

garethellis0 commented on src/vision/scripts/imgPublisher.py:
```diff
@@ -0,0 +1,81 @@
+motion = BasicMotionDetector(minArea=500)
+total = 0
+
+ImgStitcher = rospy.Publisher('ImgStitcher', Image, queue_size=10) 
```
we may want to change the name here to something more descriptive `/image_stitcher/stitched_image`

Coronal-Halo replied:
Fixed

garethellis0 reviewed (changes requested):
- Looks like you have some `.pyc` files in here, which are "compiled" python files that we don't want. To remove them, you can either use `git rm` on all the files, then `git commit` and `git push` to push your deletions, or you can just delete the files, and then "add" the deleted files with `git add` (and then commit and push them). That might be a bit confusing, so feel free to shoot me a message and I can help show you how. Pretty straight forward after you do it once
- To avoid accidentally commiting `.pyc` files in the future, you can make them "invisible" to git (so that `git add` won't add them, by adding the line `*.pyc` to the `.gitignore` file in the root directory of this repo (that `IARRC-2017` folder 

Please reply to each of my comments when you resolve them. It can be something as simple as just "fixed", or if you did something specific to fix it, put that down. My comments might be wrong, so just reply to them explaining why they are (if they are).

Overall looks pretty good, just some small changes and deletions to clean it up.

Coronal-Halo committed:
removed .pyc files

Coronal-Halo committed:
removed .pyc files

Coronal-Halo committed:
Fixed some problems

added build and run dependency of sensor_msgs
removed 2 print lines in image_Stitcher.py for clearer outputs
deleted "anonymous = True" in ros.init_node function in image_Stitcher.py
deleted 3 example files
Tested image_stitcher.py using RVIZ and verified the program actually publishes
	the stitched video/images

garethellis0 commented:
Alright, everything looks good to go! :+1: :tada: 

Go ahead and press `Squash and merge`, which will squash all the commits on this PR into 1 large PR and  merge it to master. *By doing this you also agree that you're confident that you've done your best to ensure that the code performs as expected and is free of bugs*
